mlflow/server/js/src/components/modals/RenameRunModal.js

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import RenameRunFormView from './RenameRunFormView';
+
+import Utils from '../../utils/Utils';
+import ReactModal from 'react-modal';
+
+import { setTagApi, getUUID } from '../../Actions';
+import { withRouter } from 'react-router-dom';
+import Routes from "../../Routes";
+
+
+const modalStyles = {
+  content: {
+    top: '50%',
+    left: '50%',
+    right: 'auto',
+    bottom: 'auto',
+    marginRight: '-50%',
+    transform: 'translate(-50%, -50%)'
+  },
+  overlay: {
+    backgroundColor: 'rgba(33, 37, 41, .75)'
+  }
+};
+
+class RenameRunModal extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.onRequestCloseHandler = this.onRequestCloseHandler.bind(this);
+  }
+
+  state = {
+    isSubmittingState: false,
+  }
+
+  static propTypes = {
+    open: PropTypes.bool,
+    experimentId: PropTypes.number.isRequired,
+    runUuid: PropTypes.string.isRequired,
+    runName: PropTypes.string.isRequired,
+    onClose: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    history: PropTypes.object.isRequired,
+  }
+
+  /**
+   * Form-submission handler with method signature as prescribed by Formik.
+   * See https://github.com/jaredpalmer/formik#how-form-submission-works for an explanation
+   * of how / when this method is called.
+   */
+  handleSubmit = (
+    values,
+    {
+      setSubmitting,
+    }) => {
+    const { newRunName } = values;
+    this.setState({ isSubmittingState: true });
+    const tagKey = Utils.runNameTag;
+    const setTagRequestId = getUUID();
+    return this.props.dispatch(
+      setTagApi(this.props.runUuid, tagKey, newRunName, setTagRequestId)).catch((err) => {
+        // TODO: remove alert, redirect to an error page on failed requests once one exists
+        // eslint-disable-next-line no-alert
+        alert("Unable to rename run, got error '" + err + "'. Redirecting to parent experiment " +
```

Comment:
For the TODO, this seems a reasonable pattern for handling a failed submit (under `Describe alternatives you've considered`)
https://github.com/jaredpalmer/formik/issues/711

I wouldn't redirect on error, because it loses the user's input.

Reply:
Unfortunately for now the error message is a generic "Failed to rename run, error: XHR failed" but it seems possible to improve by adding logic [here](https://github.com/mlflow/mlflow/blob/4047ac4b81cf8d02f2a48a7e0cb0f6098ddcc8ed/mlflow/server/js/src/Actions.js#L103). One possibility is to at least propagate/render the server error code (e.g. `"Failed to rename run. Error message: " + errorMessage + ", server error code: " + errorCode"`) for debuggability